### ISSUE TITLE
downloads pausing at near completion - requires manual resuming of .par2 files

### ISSUE THREAD

**Issue by natoriousbigg** [nzbget/nzbget#424]
Starting having this issue since upgrading to NZBGet 19.0/19.1. It seems to be occurring randomly every 30-40 downloads. Downloads would download fine until 99-100% completion but then pause for no reason. When I try to resume the downloads, it pause again after a few seconds. There's no error message to explain why the download has paused. I then found a solution. Click on the download -> click on Files -> check all files (all .par2) -> click Resume -> then click Save. Then the download completes normally. Please see the attached screenshots.

![screen shot 2017-08-06 at 1 25 46 pm](https://user-images.githubusercontent.com/3040306/29005899-311fd8b8-7aab-11e7-9e39-52f26de9bc9a.png)
![screen shot 2017-08-06 at 1 27 47 pm](https://user-images.githubusercontent.com/3040306/29005900-32a509ec-7aab-11e7-9ede-2cf4cf3eb4bb.png)

**Comment by hugbug** [comment]
> There's error message to explain why the download has paused.

What was it?

**Comment by natoriousbigg** [comment]
Sorry I meant "There's NO error message to explain why the download has paused."

**Comment by hugbug** [comment]
Download is showed as paused because all files inside it are paused. They are paused because they are par2-files. That's normal.

What isn't normal however is that the download hasn't transited into post-processing stage.

Please post the log - click on download, then on button "Log", then "Save to file", upload file to pastebin.com.

**Comment by natoriousbigg** [comment]
I've uploaded the log here right after it paused: https://pastebin.com/0b85ShAW

FYI I have direct unpacking enable. Looking at the log, looks like everything downloaded fines and extracted fine. When check the download folder, it's already moved to the completed folder and is already extracted inside the _unpack folder. However, postprocess isn't being triggered.

EDIT: Inside the download folder all .rar files have been deleted also.

**Comment by hugbug** [comment]
Please open in browser http://<IP>:6789/jsonrpc/listgroups and post the result via pastebin.com or send to me to nzbget@gmail.com.

**Comment by natoriousbigg** [comment]
Here it is: https://pastebin.com/Pp0cVAVY

**Comment by hugbug** [comment]
Thank you. I now have to analyse the source code and will try to reproduce this.

**Comment by natoriousbigg** [comment]
Great. I can upload the nzb that's causing the issue too.

**Comment by hugbug** [comment]
I think it's not related to a particular nzb. Can you please "Download again" from history this nzb to check this?

**Comment by natoriousbigg** [comment]
It was reproducible when I downloaded it again with another instance of nzbget which has a similar setup including the same usenet servers.

**Comment by hugbug** [comment]
Then please send me the test-nzb - nzbget@gmail.com.

**Comment by natoriousbigg** [comment]
Sent to your email!

**Comment by natoriousbigg** [comment]
This is a separate issue, but with direct unpacking enabled the downloads are being moved to the final destination before postprocessing is complete. Not sure if this is intended but I prefer to have it moved only after postprocessing is complete.

**Comment by hugbug** [comment]
The nzb contains par2-files filled with zeros (completely damaged files). That confuses NZBGet.

Will be fixed soon.

> the downloads are being moved to the final destination before postprocessing is complete

Files are always unpacked directly into destination to avoid costly moving (if on different drives). That's applies to unpack during post-processing as well as to direct unpack.

**Comment by desimaniac** [comment]
Thanks @hugbug. I am having the exact same issue. 100% files downloaded, but nzb itself is paused, due to the pars.

**Comment by hugbug** [comment]
Please send me nzb-files which are causing this - nzbget@gmail.com.

**Comment by desimaniac** [comment]
Done. Thanks

**Comment by hugbug** [comment]
@desimaniac: the same issue - the nzb contains par2-files filled with zeros (completely damaged files). That confuses NZBGet. Will be fixed soon.

**Comment by hugbug** [comment]
Fixed.

**Comment by desimaniac** [comment]
Thanks!

**Comment by qazero** [comment]
Are you sure this has been resolved? I'm experiencing the same issues. Logs provided at the link below.

https://pastebin.com/eANPKdP4

I am running verson 21.0

**Comment by ghost** [comment]
I'm also having this issue.
